packaging/osx/buildpackage.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# OpenRA packaging script for macOS
+
+command -v make >/dev/null 2>&1 || { echo >&2 "The OpenRA mod template requires make."; exit 1; }
+command -v python >/dev/null 2>&1 || { echo >&2 "The OpenRA mod template requires python."; exit 1; }
+command -v curl >/dev/null 2>&1 || { echo >&2 "The OpenRA mod template requires curl."; exit 1; }
+
+if [ $# -ne "2" ]; then
+	echo "Usage: `basename $0` tag outputdir"
+    exit 1
+fi
+
+PACKAGING_DIR=$(python -c "import os; print(os.path.dirname(os.path.realpath('$0')))")
+TEMPLATE_ROOT="${PACKAGING_DIR}/../../"
+
+# shellcheck source=mod.config
+. "${TEMPLATE_ROOT}/mod.config"
+
+if [ -f "${TEMPLATE_ROOT}/user.config" ]; then
+	# shellcheck source=user.config
+	. "${TEMPLATE_ROOT}/user.config"
+fi
+
+# Set the working dir to the location of this script
+cd ${PACKAGING_DIR}
+
+TAG="$1"
+OUTPUTDIR="$2"
+BUILTDIR="$(pwd)/build"
+
+modify_plist() {
+	sed "s|$1|$2|g" "$3" > "$3.tmp" && mv "$3.tmp" "$3"
+}
+
+echo "Building launcher"
+curl -s -L  -o "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}" -O "${PACKAGING_OSX_LAUNCHER_SOURCE}" || exit 3
+
+unzip -qq -d "${BUILTDIR}" "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}"
+rm "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}"
+
+modify_plist "{DEV_VERSION}" "${TAG}" "${BUILTDIR}/OpenRA.app/Contents/Info.plist"
+modify_plist "{FAQ_URL}" "${PACKAGING_FAQ_URL}" "${BUILTDIR}/OpenRA.app/Contents/Info.plist"
+echo "Building core files"
+
+pushd ${TEMPLATE_ROOT} > /dev/null
+
+if [ ! -f "${ENGINE_DIRECTORY}/Makefile" ]; then
+	echo "Required engine files not found."
+	echo "Run \`make\` in the mod directory to fetch and build the required files, then try again.";
+	exit 1
+fi
+
+pushd ${ENGINE_DIRECTORY} > /dev/null
+make osx-dependencies
+make core SDK="-sdk:4.5"
+make version VERSION="${TAG}"
+make install-core gameinstalldir="/Contents/Resources/" DESTDIR="${BUILTDIR}/OpenRA.app"
+popd > /dev/null
+popd > /dev/null
+
+# Remove default mods
+pushd "${BUILTDIR}/OpenRA.app/Contents/Resources/mods" > /dev/null
+rm -rf "ra" "cnc" "d2k" "modcontent"
+popd > /dev/null
+
+# Add mod files
+cp -r "${TEMPLATE_ROOT}/mods/"* "${BUILTDIR}/OpenRA.app/Contents/Resources/mods"
+cp "${MOD_ID}.icns" "${BUILTDIR}/OpenRA.app/Contents/Resources/OpenRA.icns"
+
+pushd "${BUILTDIR}" > /dev/null
+mv "OpenRA.app" "${PACKAGING_OSX_APP_NAME}"
```

Comment:
We should append `.app` to `PACKAGING_OSX_APP_NAME` if it doesn't already exist.
Someone is bound to not include the `.app` someday and doing this will make their life and ours easier.